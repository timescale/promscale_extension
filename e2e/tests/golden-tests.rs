use crate::common::{exec_sql_script, run_postgres};
use insta::assert_snapshot;
use std::env;
use test_generator::test_resources;
use testcontainers::clients;

mod common;

#[test_resources("testdata/*.sql")]
fn golden_test(resource: &str) {
    let docker = clients::Cli::default();
    let node = run_postgres(&docker);
    let query_result = exec_sql_script(&node, resource);

    assert_snapshot!(resource, query_result);
}
