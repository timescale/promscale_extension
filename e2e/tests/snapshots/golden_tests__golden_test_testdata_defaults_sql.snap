---
source: e2e/tests/golden-tests.rs
expression: query_result
---
\set ON_ERROR_STOP 1
create extension promscale;
CREATE EXTENSION
\echo make sure the view returns what we expect it to
make sure the view returns what we expect it to
select key, value
from _prom_catalog.default_default dd
except
select key, value
from
(
    values
    ('chunk_interval'           , (INTERVAL '8 hours')::text),
    ('retention_period'         , (90 * INTERVAL '1 day')::text),
    ('metric_compression'       , (exists(select 1 from pg_catalog.pg_proc where proname = 'compress_chunk')::text)),
    ('trace_retention_period'   , (30 * INTERVAL '1 days')::text),
    ('ha_lease_timeout'         , '1m'),
    ('ha_lease_refresh'         , '10s')
) x(key, value)
;
 key | value 
-----+-------
(0 rows)

\echo make sure the getter returns the expected value
make sure the getter returns the expected value
select _prom_catalog.get_default_value('chunk_interval') = (INTERVAL '8 hours')::text;
 ?column? 
----------
 t
(1 row)

\echo setting the default value to the same as the default default should be a noop
setting the default value to the same as the default default should be a noop
select _prom_catalog.set_default_value('chunk_interval', (INTERVAL '8 hours')::text);
 set_default_value 
-------------------
 
(1 row)

select count(*) = 0 from _prom_catalog.default d where d.key = 'chunk_interval';
 ?column? 
----------
 t
(1 row)

\echo overriding the default default should land a new row in the default table
overriding the default default should land a new row in the default table
select _prom_catalog.set_default_value('chunk_interval', (INTERVAL '99 hours')::text);
 set_default_value 
-------------------
 
(1 row)

select count(*) = 1 from _prom_catalog.default d where d.key = 'chunk_interval';
 ?column? 
----------
 t
(1 row)

\echo make sure the getter returns the new value
make sure the getter returns the new value
select _prom_catalog.get_default_value('chunk_interval') = (INTERVAL '99 hours')::text;
 ?column? 
----------
 t
(1 row)

\echo setting the default value BACK to the same as the default default should remove the row from the default table
setting the default value BACK to the same as the default default should remove the row from the default table
select _prom_catalog.set_default_value('chunk_interval', (INTERVAL '8 hours')::text);
 set_default_value 
-------------------
 
(1 row)

select count(*) = 0 from _prom_catalog.default d where d.key = 'chunk_interval';
 ?column? 
----------
 t
(1 row)


